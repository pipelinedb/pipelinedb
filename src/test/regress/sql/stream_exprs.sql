CREATE CONTINUOUS VIEW test_stream_exprs AS SELECT b::boolean, t::text, n::numeric FROM test_exprs_stream;

INSERT INTO test_exprs_stream (b, t, n) VALUES (true and true, substring('string!', 1, 3), 1.2 + 100.0001);
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 < 2, 'first' || 'second', 100 % 2 * log(2, 3));
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 is null, lower('UPPER'), pow(6!::float8 + 4, 3));
INSERT INTO test_exprs_stream (b, t, n) VALUES (true and true and true and false, trim(both 'x' from 'xTomxx'), pi() * mod(1 + 3, 4 * 8));
INSERT INTO test_exprs_stream (b, t, n) VALUES ('t', overlay('Txxxxas' placing 'hom' from 2 for 4), acos(1) * trunc(4.003));
INSERT INTO test_exprs_stream (b, t, n) VALUES ('on' and true, substring(upper('lower'), 1, 2), 42.82 - @(-12));
INSERT INTO test_exprs_stream (b, t, n) VALUES ('on' or 'off', btrim('xyxtrimyyx', 'xy'), 2^32);
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 * 2 > 2 * 3, md5(md5(md5('md5 me three times!'))), |/(64));
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 + 4 * 3 - 14 < 1.2, 'true', pow(2::float8, 32::numeric));
INSERT INTO test_exprs_stream (b, t, n) VALUES (true and not false or true, substring('left', 1, 2) || substring('right', 1, 2), 1 << 4 << 4);
INSERT INTO test_exprs_stream (b, t, n) VALUES (null and true, substring('x' || 'blah' || 'x', 2, 4), (100 + 24) * (16 * 3 + 4) - (16 << 2 >> 2));

-- Nonexistent fields should default to null
INSERT INTO test_exprs_stream (b) VALUES (false);
INSERT INTO test_exprs_stream (t) VALUES ('text!');
INSERT INTO test_exprs_stream (n) VALUES (1 - 1);

SELECT * FROM test_stream_exprs ORDER BY b, t, n;

DROP CONTINUOUS VIEW test_stream_exprs;
