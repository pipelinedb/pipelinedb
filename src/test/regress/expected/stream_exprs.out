SET debug_sync_stream_insert = 'on';
CREATE CONTINUOUS VIEW test_stream_exprs AS SELECT b::boolean, t::text, n::numeric FROM test_exprs_stream;
ACTIVATE test_stream_exprs;
INSERT INTO test_exprs_stream (b, t, n) VALUES (true and true, substring('string!', 1, 3), 1.2 + 100.0001);
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 < 2, 'first' || 'second', 100 % 2 * log(2, 3));
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 is null, lower('UPPER'), pow(6!::float8 + 4, 3));
INSERT INTO test_exprs_stream (b, t, n) VALUES (true and true and true and false, trim(both 'x' from 'xTomxx'), pi() * mod(1 + 3, 4 * 8));
INSERT INTO test_exprs_stream (b, t, n) VALUES ('t', overlay('Txxxxas' placing 'hom' from 2 for 4), acos(1) * trunc(4.003));
INSERT INTO test_exprs_stream (b, t, n) VALUES ('on' and true, substring(upper('lower'), 1, 2), 42.82 - @(-12));
INSERT INTO test_exprs_stream (b, t, n) VALUES ('on' or 'off', btrim('xyxtrimyyx', 'xy'), 2^32);
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 * 2 > 2 * 3, md5(md5(md5('md5 me three times!'))), |/(64));
INSERT INTO test_exprs_stream (b, t, n) VALUES (1 + 4 * 3 - 14 < 1.2, 'true', pow(2::float8, 32::numeric));
INSERT INTO test_exprs_stream (b, t, n) VALUES (true and not false or true, substring('left', 1, 2) || substring('right', 1, 2), 1 << 4 << 4);
INSERT INTO test_exprs_stream (b, t, n) VALUES (null and true, substring('x' || 'blah' || 'x', 2, 4), (100 + 24) * (16 * 3 + 4) - (16 << 2 >> 2));
-- Nonexistent fields should default to null
INSERT INTO test_exprs_stream (b) VALUES (false);
INSERT INTO test_exprs_stream (t) VALUES ('text!');
INSERT INTO test_exprs_stream (n) VALUES (1 - 1);
DEACTIVATE test_stream_exprs;
SELECT * FROM test_stream_exprs ORDER BY b, t, n;
 b |                t                 |         n          
---+----------------------------------+--------------------
 f | 497de41c131f736ed205358514b1e954 |                  8
 f | Tom                              |   12.5663706143592
 f | upper                            |          379503424
 f |                                  |                   
 t | firstsecond                      | 0.0000000000000000
 t | leri                             |                256
 t | LO                               |              30.82
 t | str                              |           101.2001
 t | Thomas                           |                  0
 t | trim                             |         4294967296
 t | true                             |         4294967296
   | blah                             |               6432
   | text!                            |                   
   |                                  |                  0
(14 rows)

DROP CONTINUOUS VIEW test_stream_exprs;
